jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -6555,8 +6555,17 @@ private static char convEndian(boolean isBigEndian, char value) {
 	public native <V> void putValue(Object obj, long offset, Class<?> clz, V value);
 	public native <V> V uninitializedDefaultValue(Class<?> clz);
 	public native <V> long valueHeaderSize(Class<V> clz);
+
+	/**
+	 * Determines whether a class is a flattened array
+	 *
+	 * @param clz the class to check
+	 * @return boolean value indicating whether the class is a flattened array
+	 */
 	public native boolean isFlattenedArray(Class<?> clz);
 
+	private native boolean isFieldFlattenedHelper(Field field);
```

Comment:
not sure we need the helper, `isFlattened` can simply be a native that does all the checks

Reply:
Are we allowed to change the public API `isFlattened()` from non-native to native ?

Reply:
When I spoke to @hangshao0 about that I believe he told me that isFlattened cannot be native or the API would be incorrect.

Reply:
Functionally speaking there is no difference so long as it remains a public method.

Reply:
We already do something similar in j.l.Object where `hashCode()` is a native in OpenJDK but ours is not.

Reply:
addressed in e295c84